File: ring-android/app/src/main/java/cx/ring/fragments/PluginHandlersListFragment.java
package cx.ring.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import cx.ring.daemon.Ringservice;
import cx.ring.databinding.FragPluginHandlersListBinding;
import cx.ring.plugins.PluginUtils;
import cx.ring.settings.pluginssettings.PluginDetails;
import cx.ring.settings.pluginssettings.PluginsListAdapter;
import cx.ring.utils.ConversationPath;


public class PluginHandlersListFragment extends Fragment implements PluginsListAdapter.PluginListItemListener {
    public static final String TAG = "PluginListHandlers";
    private FragPluginHandlersListBinding binding;
    private PluginsListAdapter mAdapter;
    private ConversationPath mPath;


    @Nullable
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragPluginHandlersListBinding.inflate(inflater, container, false);

        binding.handlerList.setHasFixedSize(true);
        mAdapter = new PluginsListAdapter(PluginUtils.getChatHandlersDetails(binding.handlerList.getContext(), mPath.getAccountId(), mPath.getConversationId()), this);
        binding.handlerList.setAdapter(mAdapter);
        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle args = getArguments();
        if (args != null) {
            mPath = ConversationPath.fromBundle(args);
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.chatPluginsToolbar.setVisibility(View.VISIBLE);
        binding.chatPluginsToolbar.setOnClickListener(v -> {
            Fragment fragment = getParentFragment();
            if (fragment instanceof ConversationFragment) {
                ConversationFragment parent = (ConversationFragment) fragment;
                parent.hidePluginListHandlers();
            }
        });
    }

    public static PluginHandlersListFragment newInstance(String accountId, String peerId) {
        PluginHandlersListFragment fragment = new PluginHandlersListFragment();

        Bundle args = ConversationPath.toBundle(accountId, peerId);

        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onPluginItemClicked(PluginDetails pluginDetails) {
        Ringservice.toggleChatHandler(pluginDetails.getmHandlerId(), mPath.getAccountId(), mPath.getConversationId(), pluginDetails.isEnabled());
    }

    @Override
    public void onPluginEnabled(PluginDetails pluginDetails) {
        Ringservice.toggleChatHandler(pluginDetails.getmHandlerId(), mPath.getAccountId(), mPath.getConversationId(), pluginDetails.isEnabled());
    }
}

File: jami-android/app/src/main/java/cx/ring/fragments/PluginHandlersListFragment.kt
/*
 *  Copyright (C) 2004-2025 Savoir-faire Linux Inc.
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
package cx.ring.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import cx.ring.account.AccountEditionFragment
import cx.ring.databinding.FragExtensionHandlersListBinding
import cx.ring.extensions.ExtensionUtils
import cx.ring.settings.extensionssettings.ExtensionDetails
import cx.ring.settings.extensionssettings.ExtensionsListAdapter
import cx.ring.settings.extensionssettings.ExtensionsListAdapter.ExtensionListItemListener
import cx.ring.utils.ConversationPath
import net.jami.daemon.JamiService

class ExtensionHandlersListFragment : Fragment(), ExtensionListItemListener {
    private var binding: FragExtensionHandlersListBinding? = null
    private lateinit var mPath: ConversationPath

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mPath = ConversationPath.fromBundle(requireArguments())!!
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return FragExtensionHandlersListBinding.inflate(inflater, container, false).also { b ->
            b.handlerList.setHasFixedSize(true)
            b.handlerList.adapter = ExtensionsListAdapter(
                ExtensionUtils.getChatHandlersDetails(b.handlerList.context, mPath.accountId, mPath.conversationId.removePrefix("swarm:")), this, "")
            binding = b
        }.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding!!.chatExtensionsToolbar.visibility = View.VISIBLE
        binding!!.chatExtensionsToolbar.setOnClickListener { v: View? ->
            val fragment = parentFragment
            if (fragment is ConversationFragment) {
                fragment.hideExtensionListHandlers()
            }
        }
    }

    override fun onExtensionItemClicked(extensionDetails: ExtensionDetails) {
        JamiService.toggleChatHandler(extensionDetails.handlerId, mPath.accountId, mPath.conversationId.removePrefix("swarm:"), extensionDetails.isEnabled)
    }

    override fun onExtensionEnabled(extensionDetails: ExtensionDetails) {
        JamiService.toggleChatHandler(extensionDetails.handlerId, mPath.accountId, mPath.conversationId.removePrefix("swarm:"), extensionDetails.isEnabled)
    }

    companion object {
        val TAG = ExtensionHandlersListFragment::class.simpleName!!
        fun newInstance(accountId: String, peerId: String): ExtensionHandlersListFragment {
            val fragment = ExtensionHandlersListFragment()
            fragment.arguments = ConversationPath.toBundle(accountId, peerId)
            return fragment
        }
    }
}